src/libAtomVM/nifs.c

```diff
@@ -1284,12 +1284,40 @@ static term nif_erlang_spawn(Context *ctx, int argc, term argv[])
 static term nif_erlang_send_2(Context *ctx, int argc, term argv[])
 {
     UNUSED(argc);
+    term dest = argv[0];
+    GlobalContext *glb = ctx->global;
+    struct ListHead *processes_table_list = synclist_rdlock(&glb->processes_table);
+
+    Context *p = NULL;
```

Comment:
it looks like we are using this variable only in the `} else if (term_is_atom(dest)) {` branch, so let's declare it there.